app/validators/rules/rule_validator.py

```diff
@@ -189,10 +189,19 @@ def _validate_option_label_from_value_operator(self, operator):
         Validate the referenced answer id in `option-label-from-value` exists
         """
         answer_id = operator[next(iter(operator))][1]
-        if answer_id not in self.questionnaire_schema.answers_with_context:
+        answers = self.questionnaire_schema.answers_with_context
+        if answer_id not in answers:
             self.add_error(
                 ValueSourceValidator.ANSWER_REFERENCE_INVALID, identifier=answer_id
             )
+            return
+        if not any(
```

Comment:
Some spacing would help with readability, alternately this could just be `elif`.